modules/serverpod_auth/serverpod_auth_server/lib/src/endpoints/status_endpoint.dart

```diff
@@ -16,9 +16,53 @@ class StatusEndpoint extends Endpoint {
     return userId != null;
   }
 
-  /// Signs out a user.
+  /// **[Deprecated]** Signs out a user from all sessions.
+  /// Use `signOutDevice` for the current session or `signOutAllDevices` for all sessions.
+  @Deprecated(
+      'Use signOutDevice for the current session or signOutAllDevices for all sessions. This method will be removed in future releases.')
   Future<void> signOut(Session session) async {
-    await UserAuthentication.signOutUser(session);
+    switch (AuthConfig.current.legacyUserSignOutBehavior) {
+      case SignOutOption.currentDevice:
+        var authInfo = await session.authenticated;
+        var authKeyId = int.tryParse(authInfo?.authId ?? '');
+        if (authKeyId == null) return;
+        await UserAuthentication.revokeAuthKey(
+          session,
+          authKeyId: authKeyId,
+        );
+        break;
+      case SignOutOption.allDevices:
+        var authInfo = await session.authenticated;
+        var userId = authInfo?.userId;
+        if (userId == null) return;
+        await UserAuthentication.signOutUser(
+          session,
+          userId: userId,
+        );
+        break;
+    }
+  }
+
+  /// Signs out a user from the current session only.
+  Future<void> signOutDevice(Session session) async {
+    var authInfo = await session.authenticated;
+    var authKeyId = int.tryParse(authInfo?.authId ?? '');
+    if (authKeyId == null) return;
+    await UserAuthentication.revokeAuthKey(
```

Comment:
Suggestion: add a blank line for readability after the return and return the call to this method.

Same for the other method.